src/AeroSharp/DataAccess/MapAccess/MapOperator.cs

```diff
@@ -74,4 +74,19 @@ public Task PutAsync(string recordKey, string bin, TKey mapKey, TValue value, Ca
 
     public Task DeleteAsync(string recordKey, CancellationToken cancellationToken) =>
         _recordOperator.DeleteAsync(recordKey, _writeConfiguration, cancellationToken);
+
+    public async Task<KeyValuePair<TKey, TValue>> GetByRankAsync(
+        string recordKey,
+        string bin,
+        int rank,
+        CancellationToken cancellationToken)
+    {
+        var operation = MapOperations.GetByRank(bin, rank);
+
+        var record = await _recordOperator.OperateAsync(recordKey, operation, _writeConfiguration, cancellationToken);
+
+        return record is null
+            ? throw new RecordNotFoundException($"No record found in map for rank \"{rank}\".")
+            : _mapParser.Parse<TKey, TValue>(record, bin);
```

Comment:
Since we're doing this stuff a few times in this file, it's probably worth pulling this into a `private` method, maybe `ValidateAndParseRecord`?

Reply:
They say imitation is the sincerest form of flattery